3scale/apicast-nginx-module
# Fix memory leak

Update the Dockerfile to use 1.21.4.3 and fix a memory leak

## Verification steps
* Checkout this branch
* Build a new container image
```
docker build -t 3scale/apicast-nginx-module .
```
* Start a new container
```
docker run --rm -v .:/opt -it 3scale/apicast-nginx-module bash
```
* Download and compile
```
make download
make patch
make compile
make openssl
```
* Generate certs
```
./create-certs.sh
./create-intermediate-certs.sh
```
* Start nginx server
```
cd test
valgrind --tool=memcheck --leak-check=full --keep-debuginfo=yes --show-possibly-lost=no --gen-suppressions=all --suppressions=../valgrind.suppress openresty -c /opt/test/mtls.conf
```
* Open another terminal and login into the container
* Send a test request
```
curl localhost:8080 
```
* Wait for the request to finish and back to the first terminal
* Stop nginx server
```
CTRL-C
```
* Check valgrind result
```
==10641== LEAK SUMMARY:                                                        
==10641==    definitely lost: 0 bytes in 0 blocks                              
==10641==    indirectly lost: 0 bytes in 0 blocks                              
==10641==      possibly lost: 128 bytes in 1 blocks                            
==10641==    still reachable: 12,041 bytes in 142 blocks                       
==10641==         suppressed: 442,376 bytes in 4 blocks                        
==10641== Reachable blocks (those to which a pointer was found) are not shown. 
==10641== To see them, rerun with: --leak-check=full --show-leak-kinds=all     
```
